F1WM/Services/CachingService.cs

```diff
@@ -1,25 +1,50 @@
-using System;
 using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Threading;
 
 namespace F1WM.Services
 {
 	public class CachingService : ICachingService
 	{
 		private readonly IMemoryCache cache;
+		private static CancellationTokenSource _resetCacheToken = new CancellationTokenSource();
 
 		public T Get<T>(string key)
 		{
 			return cache.Get<T>(key);
 		}
 
-		public T Set<T>(string key, T value, MemoryCacheEntryOptions options)
+		public T TryGetCacheValue<T>(string key) where T : class
+		{
+			if (cache.TryGetValue(key, out T cacheValue))
+			{
+				return cacheValue;
+			}
+			return null;
+		}
+
+		public void Set<T>(string key, T value, TimeSpan absoluteExpirationRelativeToNow)
 		{
-			return cache.Set<T>(key, value, options);
+			var options = new MemoryCacheEntryOptions().SetPriority(CacheItemPriority.Normal).SetSize(1).SetAbsoluteExpiration(absoluteExpirationRelativeToNow);
+			options.AddExpirationToken(new CancellationChangeToken(_resetCacheToken.Token));
+			cache.Set<T>(key, value, options);
+		}
+
+		public void DisposeCache()
```

Comment:
You might want to add methods for disposing specific cache keys (or groups of keys, using common `CancellationTokenSource` - [example](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-3.1#cache-dependencies-1)). This way we wouldn't need to drop whole cache when only a portion of data is affected, e.g. right now we drop the cache when broadcasts info changes.

Reply:
That's good idea but I would create such endpoint in lets say second iteration of our caching. For now the only CRUD in API is Broadcasts, so gain would not be so great. But with another endpoints we can add possibility to add specific keys and handle cache in 100% legit manner :)